### ISSUE TITLE
internal/contour: convert HoldoffNotifier debug logging to a metric

### ISSUE THREAD

**Issue by davecheney** [heptio/contour#591]
Issue #577 introduced a bunch of logging for k8s watcher update events. This should be converted to a prometheus metric or dropped before 0.6 final.

**Comment by alexbrand** [comment]
Looking at the log statements in the HoldoffNotifier, it seems like we could produce two metrics:
- Timestamp of the last update (or time since last update)
- Time until next update

Is this in line with what you had in mind @davecheney?

My initial thoughts here is that, given that the delay window is so small (max of 500ms), I am not sure if there's value in having these metrics.

**Comment by rosskukulinski** [comment]
I think the more helpful thing would be a prometheus histogram value for the amount of time for each update.

Alternatively, we should have the `timestamp` of last update (not time until next update) based on [this doc](https://prometheus.io/docs/practices/instrumentation/#timestamps-not-time-since)

I agree with @davecheney that this should land before 0.6.0 ships.

**Comment by davecheney** [comment]
I think 

a time of last update (the longest holdoff even is 600ms, which is below the prom reporting threshold, it probably won't be useful
b count of updates received 
c count of updates deferred
d count of updates processed immediately
e count of actual update calls

c + d should equal b and b - e will tell us how effective the holdoff timer is

a. is just for sanity checking.

**Comment by stevesloka** [comment]
@davecheney so for the count pieces, I'd have to add counters to the OnAdd, OnDelete, OnChange events? Do you have a good way to determine those metrics?

**Comment by stevesloka** [comment]
Had a quick chat in Slack and decided to just implement the timestamp metric.
